inference-engine/tests/unit/vpu/middleend_tests/passes_tests/reshape_before_conv.cpp

```diff
@@ -0,0 +1,550 @@
+    //  [FirstInput] -> (Reshape) -> [FirstInputAfterReshape] ->| -> (StubConv) -> [FirstOutputBeforeReshape] -> (Reshape) -> [FirstOutput]
+    //
+    //                                [Fake]                    ->|
+    //                                [SecondWeights]           ->|
+    //                                [Fake]                    ->|
+    //  [SecondInput] -> (Reshape) -> [SecondInputAfterReshape] ->| -> (StubConv) -> [SecondOutputBeforeReshape] -> (Reshape) -> [SecondOutput]
+    //
+    DimValues inDims = DimValues{ {Dim::N, 1}, {Dim::C, 608}, {Dim::H, 34}, {Dim::W, 60} };
+    DimValues firstOutDims = DimValues{ {Dim::N, 1}, {Dim::C, 10}, {Dim::H, 34}, {Dim::W, 60} };
+    DimValues secondOutDims = DimValues{ {Dim::N, 1}, {Dim::C, 128}, {Dim::H, 34}, {Dim::W, 60} };
+
+    const auto firstInput = InitInputData(inDims);
+    const auto secondInput = InitInputData(inDims);
+    _model->attrs().set<int>("numInputs", 2);
+    const auto firstOutput = InitOutputData(firstOutDims);
+    const auto secondOutput = InitOutputData(secondOutDims);
+
+    InitConvStage(firstInput, firstOutput);
+    InitConvStage(secondInput, secondOutput);
+
+    ASSERT_NO_THROW(Compile());
+
+    const Data fake = _model->addFakeData();
+    const Data firstWeights = InitNewData(
+            DimValues{ {Dim::N, 10}, {Dim::C, 608}, {Dim::H, 1}, {Dim::W, 1} },
+            "TestConvolution@weights@conv");
+    const Data secondWeights = InitNewData(
+            DimValues{ {Dim::N, 128}, {Dim::C, 608}, {Dim::H, 1}, {Dim::W, 1} },
+            "TestConvolution@weights@conv");
+    const Data firstInputAfterReshape = InitNewData(
+            DimValues{ {Dim::N, 1}, {Dim::C, 608}, {Dim::H, 8}, {Dim::W, 255} },
+            "Input@input-data-after-reshape");
+    const Data secondInputAfterReshape = InitNewData(
+            DimValues{ {Dim::N, 1}, {Dim::C, 608}, {Dim::H, 8}, {Dim::W, 255} },
+            "Input@input-data-after-reshape");
+    const Data firstOutputBeforeReshape = InitNewData(
+            DimValues{ {Dim::N, 1}, {Dim::C, 10}, {Dim::H, 8}, {Dim::W, 255} },
+            "Output@output-data-before-reshape");
+    const Data secondOutputBeforeReshape = InitNewData(
+            DimValues{ {Dim::N, 1}, {Dim::C, 128}, {Dim::H, 8}, {Dim::W, 255} },
+            "Output@output-data-before-reshape");
+
+    pattern_datas.push_back(firstInput);
+    pattern_datas.push_back(secondInput);
+    pattern_datas.push_back(firstOutput);
+    pattern_datas.push_back(secondOutput);
+    pattern_datas.push_back(fake);
+    pattern_datas.push_back(fake);
+    pattern_datas.push_back(firstWeights);
+    pattern_datas.push_back(fake);
+    pattern_datas.push_back(fake);
+    pattern_datas.push_back(secondWeights);
+    pattern_datas.push_back(firstInputAfterReshape);
+    pattern_datas.push_back(firstOutputBeforeReshape);
+    pattern_datas.push_back(secondInputAfterReshape);
+    pattern_datas.push_back(secondOutputBeforeReshape);
```

Comment:
Still a lot of `push_backs` in tests. Are there any changes to get rid of them?

Reply:
fixed